src/utils/distributions.h

```diff
@@ -56,6 +56,33 @@ Eigen::VectorXd multi_normal_prec_lpdf_grid(const Eigen::MatrixXd &data,
                                             const Eigen::MatrixXd &prec_chol,
                                             double prec_logdet);
 
+/*
+ * Returns a pseudorandom multivariate normal random variable with diagonal
+ * covariance matrix
+ *
+ * @param mean   The mean of the Gaussian r.v.
+ * @param cov_diag   The diagonal covariance matrix
+ * @rng          Random number generator
+ * @return       multivariate normal r.v.
+ */
```

Comment:
New methods to sample with diagonal covariance matrix and by passing the Cholesky factorization of the precision matrix directly.